packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart

```diff
@@ -203,15 +204,34 @@ class BuildDaemonCreator {
       'lib',
       'build_script.dart',
     );
+    final String snapshot = globals.fs.path.join(
+      Cache.flutterRoot,
+      'bin',
+      'cache',
+      'build_script.snapshot',
+    );
     if (!globals.fs.isFileSync(buildScript)) {
       throwToolExit('Expected a file $buildScript to exist in the Flutter SDK.');
     }
-    // If we're missing the .packages file, perform a pub get.
-    if (!globals.fs.isFileSync(buildScriptPackages)) {
-      await pub.get(
-        context: PubContext.pubGet,
-        directory: globals.fs.file(buildScriptPackages).parent.path,
-      );
+
+    if (!globals.fs.isFileSync(snapshot)) {
+      // If we're missing the .packages file, perform a pub get.
+      if (!globals.fs.isFileSync(buildScriptPackages)) {
+        await pub.get(
+          context: PubContext.pubGet,
+          directory: globals.fs.file(buildScriptPackages).parent.path,
+        );
+      }
+
+      final ProcessResult snapshotResult = await globals.processManager.run(<String>[
+        globals.artifacts.getArtifactPath(Artifact.engineDartBinary),
+        '--snapshot=$snapshot',
```

Comment:
`--snapshot-kind=app-jit` ? or does that not make a big difference

Reply:
I'll give it a shot. Heads up, this PR is not ready for review yet. I'll need to chat with you about how to properly cache and invalidate the snapshot.